backend/frontend/templates/frontend/partial_logged_in_main_header.html

```diff
@@ -91,16 +58,10 @@
                   tabindex="-1"
                 >
                   <a
-                    href="#"
+                    href="{% url 'user_board' user.pk %}"
```

Comment:
Your Board link is linking to the wrong user. It should link to the request user's board instead.